client/src/pages/Dashboard.js

```diff
@@ -40,46 +38,60 @@ function Dashboard() {
     axios
       .get(`/settings/${localStorage.getItem("email")}`)
       .then(res => {
-        if (res.data.success) {
+        if (res.data.authenticated === false) {
+          handleLogout();
+        } else if (res.data.success) {
           setPlatforms(res.data.settings.platforms);
           setCompanies(res.data.settings.companies);
           res.data.settings.companies.forEach(company => {
             axios
               .get(`/reddit/search/new/${company}`)
               .then(res => {
-                if (res.data.success) {
+                if (res.data.authenticated === false) {
+                  handleLogout();
+                } else if (res.data.success) {
                   setMentions(res.data.submissions);
                 }
               })
               .catch(error => {
-                console.error(error)
-              })
-          })
+                console.error(error);
+              });
+          });
         }
       })
       .catch(error => {
-        console.error(error)
-      })
+        console.error(error);
+      });
   }, []);
 
   const handlePlatformChange = platform => {
     axios
       .put(`/settings/${localStorage.getItem("email")}/platform/${platform}`)
       .then(res => {
-        if (res.data.success) {
+        if (res.data.authenticated === false) {
```

Comment:
Is there a particular reason we are checking for `res.data.authenticated === false`.

I feel like this could be simplified to `if(!res.data.authenticated)`

Reply:
I had this originally but then it would be undefined on success which is a falsy value as well, which I would give a false positive on this.